app/uk/gov/hmrc/gform/des/DesConnector.scala

```diff
@@ -31,17 +37,56 @@ import uk.gov.hmrc.http.logging.Authorization
 class DesConnector(wSHttp: WSHttp, baseUrl: String, desConfig: DesConnectorConfig) {
 
   def lookupRegistration(utr: String, desRegistrationRequest: DesRegistrationRequest)(
-    implicit ex: ExecutionContext): Future[DesRegistrationResponse] = {
+    implicit ex: ExecutionContext): Future[ServiceCallResponse[DesRegistrationResponse]] = {
+
     implicit val hc = HeaderCarrier(
       extraHeaders = Seq("Environment" -> desConfig.environment),
       authorization = Some(Authorization(s"Bearer ${desConfig.authorizationToken}")))
     Logger.info(s"Des registration, UTR: '$utr', ${loggingHelpers.cleanHeaderCarrierHeader(hc)}")
-    wSHttp.POST[DesRegistrationRequest, DesRegistrationResponse](
-      s"$baseUrl${desConfig.basePath}/registration/organisation/utr/$utr",
-      desRegistrationRequest)
 
+    wSHttp
+      .doPost[DesRegistrationRequest](
+        s"$baseUrl${desConfig.basePath}/registration/organisation/utr/$utr",
+        desRegistrationRequest,
+        List.empty[(String, String)]
+      )
+      .map { httpResponse =>
+        if (httpResponse.status === 400) {
```

Comment:
this potentially will be an Upstream4xxResponse that you would have to recover on.

Reply:
`Upstream4xxResponse` cannot be used since we need to treat `code === "INVALID_UTR"` in different way from any other `code` value. All of them has `status === 400`